StoreKitUnitTests/ProductDetailsTests.swift

```diff
@@ -0,0 +1,115 @@
+            partialResult[wrapper.productIdentifier] = wrapper
+        }
+
+        expect(sk1ProductDetailss.count) == productIdentifiers.count
+        expect(sk1ProductDetailss.count) == sk2ProductDetailss.count
+
+        for sk1ProductID in sk1ProductDetailssByID.keys {
+            let sk1Product = try XCTUnwrap(sk1ProductDetailssByID[sk1ProductID])
+            let equivalentSK2Product = try XCTUnwrap(sk2ProductDetailssByID[sk1ProductID])
+
+            expect(sk1Product.productIdentifier) == equivalentSK2Product.productIdentifier
+            expect(sk1Product.localizedDescription) == equivalentSK2Product.localizedDescription
+            expect(sk1Product.price) == equivalentSK2Product.price
+            expect(sk1Product.localizedPriceString) == equivalentSK2Product.localizedPriceString
+            expect(sk1Product.productIdentifier) == equivalentSK2Product.productIdentifier
+            expect(sk1Product.isFamilyShareable) == equivalentSK2Product.isFamilyShareable
+            expect(sk1Product.localizedTitle) == equivalentSK2Product.localizedTitle
+            if sk1Product.subscriptionGroupIdentifier != nil {
+                expect(sk1Product.subscriptionGroupIdentifier) == equivalentSK2Product.subscriptionGroupIdentifier
+            } else {
+                expect(equivalentSK2Product.subscriptionGroupIdentifier).to(beNil())
+            }
+        }
+    }
+
+    func testSk1DetailsWrapsCorrectly() throws {
+        let productIdentifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        let sk1Fetcher = ProductsFetcherSK1(productsRequestFactory: ProductsRequestFactory())
+        var callbackCalled = false
+
+        sk1Fetcher.products(withIdentifiers: Set([productIdentifier])) { productDetailss in
+            callbackCalled = true
+            guard let productDetails = productDetailss.first else { fatalError("couldn't get product!") }
+
+            expect(productDetails.productIdentifier) == "com.revenuecat.monthly_4.99.1_week_intro"
+            expect(productDetails.localizedDescription) == "Monthly subscription with a 1-week free trial"
+            expect(productDetails.price.description) == "4.99"
+            expect(productDetails.localizedPriceString) == "$4.99"
+            expect(productDetails.productIdentifier) == productIdentifier
+            expect(productDetails.isFamilyShareable) == true
+            expect(productDetails.localizedTitle) == "Monthly Free Trial"
+        }
+
+        expect(callbackCalled).toEventually(beTrue())
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testSk2DetailsWrapsCorrectly() async throws {
+        guard #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) else { return }
+
+        let productIdentifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        let sk2Fetcher = ProductsFetcherSK2()
+
+        let productDetailss = try await sk2Fetcher.products(identifiers: Set([productIdentifier]))
```

Comment:
I don't know if this extra `s` is intentional, but maybe `setOfProductDetails` would be better

Reply:
lol it's not intentional, will fix